src/analyses/ai.h

```diff
@@ -94,6 +95,13 @@ class ai_domain_baset
   //
   // This computes the join between "this" and "b".
   // Return true if "this" has changed.
+
+  // Used to evaluate or simplify conditions with respect to the domain
+  virtual exprt domain_simplify (const exprt &condition,
+				 const namespacet &ns,
+				 const bool lhs = false) const {
+    return condition;
+  }
```

Comment:
Looking at the commit in isolation, I have no idea when/why to use this. It might make sense to merge it with an actual application thereof.